### PR TITLE
[Merged by Bors] - chore(Order/Bounds/Basic): Split long file into two shorter files

### PR DESCRIPTION
**Pull request by Paul-Lez**
This PR transfers some of the content from `Order/Bounds/Basic` into a new file, `Order/Bounds/Image`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/Bounds/Basic.lean [hunk 1]
```diff
@@ -1130,6 +1130,52 @@ theorem map_isLeast : IsLeast s a → IsGreatest (f '' s) (f a) :=
...
+  ⟨fun Ha => ⟨(mem_image_iff_mem_of_injective _ _ hf.injective).mp Ha.left,
+    (mem_upperBounds_image_iff hf).mp Ha.right⟩, hf.monotone.map_isGreatest⟩
+
+end StrictMono
+
+section StrictAnti
+
+variable [LinearOrder α] [Preorder β] {f : α → β} (hf : StrictAnti f) {a : α} {s : Set α}
+
+include hf
```
Comment (callesonne) [inline Mathlib/Order/Bounds/Basic.lean:1159]: I've heard its good (for readability) to include explicit args in the statements themselves. So maybe include `hf` manually in each of the statements? I am not fuzzed about it though.

**Review by callesonne — APPROVED** [review]
Apart from small comments, LGTM!

File: Mathlib/Data/Set/Operations.lean [hunk 2]
```diff
@@ -122,6 +122,12 @@ theorem mem_image (f : α → β) (s : Set α) (y : β) : y ∈ f '' s ↔ ∃ x
 theorem mem_image_of_mem (f : α → β) {x : α} {a : Set α} (h : x ∈ a) : f x ∈ f '' a :=
   ⟨_, h, rfl⟩
 
+theorem mem_image_iff_mem_of_injective {f : α → β} (s : Set α) (a : α) (hf : f.Injective) :
+    f a ∈ f '' s ↔ a ∈ s := by
+  refine ⟨fun ha => ?_, fun ha => mem_image_of_mem f ha⟩
+  obtain ⟨y, hy, hy'⟩ := (mem_image _ _ _).mp ha
+  rwa [←hf hy']
```
[the remark this answers is not in the document]
Comment (Paul-Lez) [reply Mathlib/Data/Set/Operations.lean]: Ah actually this lemma already existed in `Set/Image.lean` so I've removed my version.

**Comment by callesonne** [comment]
I would do the file splitting in a separate PR, so that reviewing this one becomes easier. Now it's hard to tell whats changed from the diff.

**Comment by Paul-Lez** [comment]
I've now done the file splitting - the new lemmas have been moved to #17595

**Review by callesonne — APPROVED** [review]
LGTM!

**Review by callesonne — CHANGES_REQUESTED** [review]
Wait, I can't remember what the added lemmas were, #17595 has the exact same diff. So are the new lemmas also in this PR?

**Comment by Paul-Lez** [comment]
> Wait, I can't remember what the added lemmas were, #17595 has the exact same diff. So are the new lemmas also in this PR?

Ah I forgot to push the lemmas in the other PR after removing them from this PR. Just added them now.

**Review by callesonne — APPROVED** [review]
Okay LGTM then!

**Comment by alexjbest** [comment]
maintainer merge

**Comment by urkud** [comment]
Thanks! :tada:
bors merge

**Comment by Paul-Lez** [comment]
bors merge

**Pull request closed without merging**